src/STAN.Client/StanConnection.cs

```diff
@@ -163,7 +163,7 @@ public class Connection : IStanConnection, IDisposable
 
         internal ProtocolSerializer ps = new ProtocolSerializer();
 
-        private StanOptions opts = null;
+        internal StanOptions opts = null;
```

Comment:
I assume  internal Connection(string stanClusterID, string clientID, StanOptions options) below is the only constructor (assuming this is one.)
Shouldn't we initialize this to new StanOptions(); and below only set it if nats options are not null.
Would make it clearer that opts will not be null.

Looks like c# has a readonly keyword. (similar to final in java?) Since you alter visibility, maybe make readonly?
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/readonly#readonly-field-example

Reply:
Good point about readonly, will fix that.

You have a point, and these are probably scars from my C days, but I don't like allocating memory where it won't necessarily be used.  Since the default constructor is private, there's only one connection constructor so can keep all allocations there imo.  To address maintainability, I'll add a comment and leave it unset to indicate programmer intent.